### ISSUE TITLE
img.shields.io blocked by privacy badger

### ISSUE THREAD

**Issue by ian-kelling** [strugee/gplenforced.org#7]
For me. Instead, it just shows a link with the alt text. I sent an email to the shields.io site owner, it seems he could put up a do not track policy page to fix this. Self-hosting is an option to consider.

**Comment by strugee** [comment]
I'd rather not self-host just in case gplenforced.org ever goes down. This seems like a bug with Privacy Badger.

**Comment by strugee** [comment]
Thanks for the report though. Leaving open as I am open to the idea of solving this downstream. Let's see what the Privacy Badger folks say.

**Comment by ghostwords** [comment]
Hi @ian-kelling, could you share a screenshot of Privacy Badger's popup on `shields.io`? Here is mine, for example:
![screenshot from 2017-05-24 18 37 14](https://cloud.githubusercontent.com/assets/794578/26428507/1346582e-40b0-11e7-9aef-1f30c443cc3b.png)

**Comment by ian-kelling** [comment]
Text equivalent of screenshot. (edit: this is when viewing gplenforced.org)

yellow (slider in middle): fonts.googleapis.com, fonts.gstatic.com
red (slider on left): img.shields.io
green(slider on right): piwik.strugee.net

**Comment by ian-kelling** [comment]
I didn't say that I contacted privacy badger. Since you suggested it, I did their in gui report that privacy badger broke the page. I don't expect them to do much of anything through. The message I sent:
"img.shields.io is important for this site, but the domain is blocked. I contacted the owner to suggest he put up a do not track page."

**Comment by ghostwords** [comment]
This is definitely a Privacy Badger bug ... first-party resources should always be left alone, plus why is Badger reporting "piwik.strugee.net" on `shields.io`?

**Comment by ian-kelling** [comment]
> why is Badger reporting "piwik.strugee.net" on shields.io?

My text screenshot is when my browser is viewing gplenforced.org, not shields.io. edit: Sorry, I didn't read your request properly.

**Comment by ian-kelling** [comment]
When viewing shields.io,

red (slider on left): ghbtns.com
yellow (slider in middle): fonts.googleapis.com, fonts.gstatic.com
green(slider on right): cloud.githubusercontent.com

**Comment by ghostwords** [comment]
Oh oops, I was confused too. Got it, the problem is on `gplenforced.org`. Let's move the discussion to https://github.com/EFForg/privacybadger/issues/1394. This ticket can be closed, I think.

**Comment by strugee** [comment]
Closing as this is an upstream Privacy Badger bug.

**Comment by ghost** [comment]
@strugee wrote:
> Closing as this is an upstream Privacy Badger bug.

I see that the Privacy Badger team has [accepted this as a bug](https://github.com/EFForg/privacybadger/issues/1394), but that doesn't make sense to me. Perhaps I have misunderstood Privacy Badger's aims, but I'll note the basis of my confusion here in case it's helpful :)

Without Privacy Badger, when a user views Site A, if that site's HTML requests resources (e.g. an SVG image) from Site X, then Site X will learn that the user's browser (which might be [uniquely identifiable](https://www.eff.org/deeplinks/2015/12/panopticlick-20-launches-featuring-new-tracker-protection-and-fingerprinting-tests) is visiting a site that requests that SVG image.

Site X might not learn that the site the user was viewing was Site A specifically (because maybe Site B and Site C also embed the same SVG), but they will learn quite a bit more about the user than would be the case if Site A self-hosted the SVG.

If the user instead runs Privacy Badger, and Privacy Badger blocks the browser's request to Site X when the user visits Site A, then Privacy Badger is surely working as desired: protecting the user from the loss of privacy described above.

**Comment by ghostwords** [comment]
You could block all third-party resources, but then half the sites will be broken to some degree. Privacy Badger tries to identify when domains perform tracking (by setting non-trivial cookies, for example) and then starts blocking those domains only after seeing them perform tracking a few times across the Web. The domain in question here shouldn't qualify for Privacy Badger's definition of tracking, which makes Badger deciding to block it a bug.

**Comment by ghost** [comment]
@ghostwords thanks for the explanation :) Presumably even a trivial cookie (or an SVG file) could be used to perform tracking, though.

I accept that *identifying when domains are performing tracking* is indeed a conversation for Privacy Badger rather than gplenforced.org , but it might be an idea for gplenforced.org to recommend its users self-host the SVG, rather than recommending they load it from a third party. I have opened a new issue for that: #11.
